libevmasm/SimplificationRules.h

```diff
@@ -87,12 +89,14 @@ class Pattern
 	using Expression = ExpressionClasses::Expression;
 	using Id = ExpressionClasses::Id;
 
-	using Builtins = dev::eth::EVMBuiltins;
+	using Builtins = dev::eth::EVMBuiltins<Pattern>;
 	static constexpr size_t WordSize = 256;
 	using Word = u256;
 
 	// Matches a specific constant value.
 	Pattern(unsigned _value): Pattern(u256(_value)) {}
+	Pattern(int _value): Pattern(u256(_value)) {}
+	Pattern(long unsigned _value): Pattern(u256(_value)) {}
```

Comment:
I wonder whether these constructors can and should be ``constexpr`` and whether the compiler could optimize more, if they were... probably, among others, that depends on what exactly ``boost::multiprecision`` does...

Actually this class needs a bit of cleanup anyways - ``m_data`` for example should be an ``optional`` and not a ``shared_ptr``, should it? (That will probably affect the ability to construct as constexpr as well)